src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -929,14 +929,6 @@ void VideoDecoder::convertAVFrameToDecodedOutputOnCPU(
           "Invalid color conversion library: " +
           std::to_string(static_cast<int>(streamInfo.colorConversionLibrary)));
     }
-    if (!preAllocatedOutputTensor.has_value()) {
-      // We only convert to CHW if a pre-allocated tensor wasn't passed. When a
-      // pre-allocated tensor is passed, it's up to the caller (typically a
-      // batch API) to do the conversion. This is more efficient as it allows
-      // batch NHWC tensors to be permuted only once, instead of permuting HWC
-      // tensors N times.
-      output.frame = MaybePermuteHWC2CHW(streamInfo.options, output.frame);
-    }
```

Comment:
This removal is actually the main change. Everything else is just (sensible) patching until tests work